dangerzone/gvisor_wrapper/entrypoint.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python3
+
+import argparse
+import json
+import os
+import shlex
+import subprocess
+import sys
+import typing
+
+# This script wraps the command-line arguments passed to it to run as an
+# unprivileged user in a gVisor sandbox.
+
+# Define flags.
+parser = argparse.ArgumentParser(
+    prog="gvisor_wrapper",
+    description="Run a command in a nested gVisor sandbox",
+    prefix_chars="-",
+    fromfile_prefix_chars=None,
+)
+flags_with_values = []
+parser.add_argument(
+    "--pre_gvisor", action="store_true", help="Run command without gVisor wrapping"
+)
+parser.add_argument(
+    "--gvisor_debug", action="store_true", help="Enable gVisor debug logging"
+)
+parser.add_argument(
+    "--gvisor_strace", action="store_true", help="Enable system call tracing in gVisor"
+)
+flags_with_values.append(
+    parser.add_argument(
+        "--gvisor_flag",
+        action="append",
+        help="Add gVisor flag, may be specified multiple times",
+    )
+)
+parser.add_argument(
+    "command",
+    nargs=argparse.PARSER,
+    help="Command to run in the sandbox (or outside if --pre_gvisor is specified)",
+)
+
+if len(sys.argv) <= 1:  # No arguments: bail.
+    print("No command line specified.", file=sys.stderr)
+    parser.print_help()
+    sys.exit(2)
+
+# We can't pass `sys.argv` directly to `parser` here because this may end up
+# processing flags from the wrapped command line that happen to match flags
+# that are defined in `parser`. For example, if the user were to somehow
+# make a file named `--pre_gvisor`, that would be pretty bad.
+# So we scan for the first argument that does not look like it is intended
+# for this entrypoint script, and we only pass this subset to `parser`.
```

Comment:
This is probably a relic from a previous iteration of Dangerzone, but we no longer let attack-controlled strings in our (tokenized) command arguments. Other than that though, I think we can simplify the argument parsing here, by doing the following:

1. Make the container isolation provider explicitly run the full `podman ... runsc ...` command. This will make things more verbose, but less magic. Also, it will make development faster, since we will have just one place to tweak gvisor arguments, and not recompile the Dangerzone image. 
2. In the entrypoint script, check if the basename of `sys.argv[0]` is  `runsc`. In that case, create the OCI config.
3. Do not handle any extra arguments in the entrypoint script, since they will provided by the isolation provider, if necessary.

Reply:
I like the way you simplified this, I think we're good.